plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -158,35 +164,59 @@ private void registerOptInStatus(boolean optedIn) {
     flushPreferences(preferences);
   }
 
+  public void sendPing(String eventName) {
+    sendPingOnShell(null, eventName);
+  }
+
+  public void sendPing(String eventName, String metadataKey) {
+    sendPingOnShell(null, eventName, metadataKey);
+  }
+
+  public void sendPing(String eventName, String metadataKey, String metadataValue) {
+    sendPingOnShell(null, eventName, metadataKey, metadataValue);
+  }
+
   /**
    * Sends a usage metric to Google Analytics.
    *
    * If the user has never seen the opt-in dialog or set the opt-in preference beforehand,
    * this method can potentially present an opt-in dialog at the top workbench level. If you
-   * are calling this method inside another modal dialog, consider using {@link #sendPing(
-   * String, String, String, Shell)} and pass the {@Shell} of the currently open modal dialog.
+   * are calling this method inside another modal dialog, consider using {@link #sendPingOnShell(
+   * Shell, String, ImmutableMap)} and pass the {@Shell} of the currently open modal dialog.
    * (Otherwise, the opt-in dialog won't be able to get input until the workbench can get input.)
    *
    * Safe to call from non-UI contexts.
    */
-  public void sendPing(String eventName, String metadataKey, String metadataValue) {
-    sendPing(eventName, metadataKey, metadataValue, null);
+  public void sendPing(String eventName, ImmutableMap<String, String> metadata) {
```

Comment:
Requiring this to be `ImmutableMap` seems overly restrictive?

Reply:
Probably. I should just copy the map.